pkg/datapath/loader/loader.go

```diff
@@ -342,16 +342,27 @@ func (l *Loader) reloadDatapath(ctx context.Context, ep datapath.Endpoint, dirs
 				}
 				return err
 			}
+		} else {
+			err := RemoveTCFilters(ep.InterfaceName(), netlink.HANDLE_MIN_EGRESS)
+			if err != nil {
+				log.WithField("device", ep.InterfaceName()).Error(err)
+			}
 		}
 	}
 
-	if ep.RequireEndpointRoute() {
-		if ip := ep.IPv4Address(); ip.IsSet() {
+	if ip := ep.IPv4Address(); ip.IsSet() {
+		if ep.RequireEndpointRoute() {
 			upsertEndpointRoute(ep, *ip.IPNet(32))
+		} else {
+			removeEndpointRoute(ep, *ip.IPNet(32))
```

Comment:
nit: Apparently we've just been ignoring errors here. For the remove case it may be run when there is no underlying route so at least for the "ENOENT" type errors we should not be logging. But seems like we should follow up to make sure we don't just hide errors here..

Reply:
I opened `good-first-issue` https://github.com/cilium/cilium/issues/15282 for it. I'll address it if noone takes it.